plugin/core/rpc.py

```diff
@@ -92,6 +97,29 @@ def send_request(self, request: Request, handler: 'Callable[[Optional[Any]], Non
             debug('unable to send', request.method)
             if error_handler is not None:
                 error_handler(None)
+            return None
+
+    def execute_request(self, request: Request, timeout: int = DEFAULT_SYNC_REQUEST_TIMEOUT) -> 'Optional[Any]':
+        """
+        Sends a request and waits for response up to timeout (default: 1 second), blocking the current thread.
+        """
+        if self.transport is None:
+            debug('unable to send', request.method)
+            return None
+
+        debug(' ==> ' + request.method)
+        self.request_id += 1
+        request_id = self.request_id
+        self.send_payload(request.to_payload(self.request_id))
+
+        current_time = start_time = time.time()
+        while current_time < start_time + timeout:
```

Comment:
Condition variables are perfect for this use-case: we have some resource updated in another thread but want to access it from the current thread. I suggest to change the while-loop here with a condition variable. Consider `git cherry-pick`ing this commit I made that already does that: 65e979a

Reply:
Thanks, this is a nice improvement! Cherry picking gave me both your suggestions.